Week-12/react-week-12/src/App.js

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import Button from "react-bootstrap/Button";
+import Card from "react-bootstrap/Card";
+import Form from "react-bootstrap/Form";
+
+function Todo({ todo, index, markTodo, removeTodo }) {
+  return (
+    <div className="todo">
+      <span style={{ textDecoration: todo.isDone ? "line-through" : "" }}>
+        {todo.text}
+      </span>
+      <div>
+        <Button variant="outline-success" onClick={() => markTodo(index)}>
+          ✓
+        </Button>{" "}
+        <Button variant="outline-danger" onClick={() => removeTodo(index)}>
+          ✕
+        </Button>
+      </div>
+    </div>
+  );
+}
+
+function FormTodo({ addTodo }) {
+  const [value, setValue] = React.useState("");
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (!value) return;
+    addTodo(value);
+    setValue("");
+  };
+
+  return (
+    <Form onSubmit={handleSubmit}>
+      <Form.Group>
+        <Form.Label>
+          <b>Add Todo</b>
+        </Form.Label>
+        <Form.Control
+          type="text"
+          className="input"
+          value={value}
+          onChange={(e) => setValue(e.target.value)}
+          placeholder="Add new todo"
+        />
+      </Form.Group>
+      <Button variant="primary mb-3" type="submit">
+        Submit
+      </Button>
+    </Form>
+  );
+}
+
+function App() {
+  const [todos, setTodos] = React.useState([
+    {
+      text: "This is a sampe todo",
+      isDone: false,
+    },
+  ]);
+
+  const addTodo = (text) => {
```

Comment:
Just a syntactical way of improving this and avoid creating a new variable

`
const addTodo = (text) => {
    const newTodos = [...todos, { text }];
    setTodos(newTodos);
  };
`

`const addTodo = (text) => setTodos([...todos, { text }]);
`